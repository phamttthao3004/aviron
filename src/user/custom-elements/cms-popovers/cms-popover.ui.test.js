import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import CmsPopoverUI from './cms-popover.ui';

describe('CmsPopoverUI component', () => {
    it('Case 1: open=false', () => {
        const props = { open: false, children: <div>Hello CmsPopoverUI</div>, className: '', identifierContainer: 'test-container-12345', identifier: 'test-56789' };
        const com = shallow(<CmsPopoverUI {...props} />);
        expect(shallowToJson(com)).toMatchSnapshot();
    });

    it('Case 2: open=true', () => {
        const props = { open: true, children: <div>Hello CmsPopoverUI</div>, className: '', identifierContainer: 'test-container-12345', identifier: 'test-56789' };
        const com = shallow(<CmsPopoverUI {...props} />);
        expect(shallowToJson(com)).toMatchSnapshot();
    });
});
