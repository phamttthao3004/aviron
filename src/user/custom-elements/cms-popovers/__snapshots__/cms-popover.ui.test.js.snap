// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CmsPopoverUI component Case 1: open=false 1`] = `""`;

exports[`CmsPopoverUI component Case 2: open=true 1`] = `
<Portal
  containerInfo={<body />}
>
  <div
    className="cms-popover-container "
    id="test-container-12345"
  >
    <div
      className="cms-popover-mask"
    />
    <div
      className="cms-popover-content"
      id="test-56789"
    >
      <div>
        Hello CmsPopoverUI
      </div>
    </div>
  </div>
</Portal>
`;
